demo/docbot/cluster_bootstrap.py

```diff
@@ -60,53 +60,54 @@ def init_index_template(client: MLClient, template_name = "nlp-template") -> Non
     returns None if template was created or is exists else raises Exception
   """
   if not client._client.indices.exists_template(name=template_name):
-    template = {
-    "index_patterns": [
-      "cohere*"
-    ],
-    "template": {
-      "settings": {
-        "number_of_shards": 1,
-        "number_of_replicas": 2,
-        "codec": "zstd_no_dict"
-        },
-      "mappings": {
-        "properties": {
-          "data": {
-            "index": False
+    if template_data is None:
+        template_data = {
+        "index_patterns": [
+        "cohere*"
+        ],
+        "template": {
+        "settings": {
+            "number_of_shards": 1,
+            "number_of_replicas": 2,
+            "codec": "zstd_no_dict"
             },
-          "ancestors": {
-            "index": False
+        "mappings": {
+            "properties": {
+            "data": {
+                "index": False
+                },
+            "ancestors": {
+                "index": False
+                }
             }
-          }
+            }
+        }
         }
-      }
-    }
 
-    response = client._client.indices.put_template(name=template_name, body=template)
+    response = client._client.indices.put_template(name=template_name, body=template_data)
     if not response["acknowledged"]:
       raise Exception("Unable to create index template.")
 
 
-def initialize_model_group(client: MLClient) -> None:
+def initialize_model_group(client: MLClient, model_group_name="Cohere_Group", data=None) -> None:
     """
     Args:
         Client: OpenSearch
     Returns:
         returns None if model group was created or is exists else raises Exception
     """
     # Later on we can edit this to allow more models. For now, we will stick to Cohere.
-    model_group_name = "Cohere_Group"
 
     response = client.get_model_group_id(model_group_name=model_group_name)
 
     # Check if the Cohere model group already exists
     if response is not None:
-        data = {
-            "name": model_group_name,
-            "description": "Public Cohere Model Group",
-            "access_mode": "public"
-        }
+        if data is None:
```

Comment:
What we can do to make data a little more modular is we can make the model group name appear in the description with f-strings so each would have the right name if we ever changed the model group name.